### PR TITLE
feat: add ens avatar support

### PR DESCRIPTION
**Pull request by bigint**
Screenshot 2021-12-24 at 10 19 25 PM

If the user don't have ens avatar it will fallback to jazzicons

### CODE REVIEW

**Comment by FoxTheSin** [comment]
Hey @bigint , we appreciate this contribution. Can you help us understand why you're using a dependency and/or would it be possible to not use the dependency in this case?

**Comment by turtlemoji** [comment]
we are updating all package and lock, so this should hold until we merge some fixes. I will add the tag.
Also we would like to use styled componentes instead of `style={}` if thats possible :)

**Comment by bigint** [comment]
> Hey @bigint , we appreciate this contribution. Can you help us understand why you're using a dependency and/or would it be possible to not use the dependency in this case?

@FoxTheSin @turtlemoji It is very difficult to add support for jazzicons + ens avatar in the same go, `@davatar/react` is a good choice to use it, even Sushiswap uses it 🙇🏻

File: package.json [hunk 1]
```diff
@@ -16,6 +16,7 @@
   },
   "dependencies": {
     "@craco/craco": "6.2.0",
+    "@davatar/react": "^1.8.1",
```
Comment (turtlemoji) [inline package.json:19]: `"@davatar/react": "^1.8.1",` should be pinned so: `"@davatar/react": "1.8.1",`

Comment (turtlemoji) [inline package.json:19]:
```suggestion
    "@davatar/react": "1.8.1",
```

**Comment by turtlemoji** [comment]
updated comments @bigint

File: src/client/components/app/ConnectWalletButton/index.tsx [hunk 2]
```diff
@@ -34,6 +35,7 @@ export const ConnectWalletButton = ({ address, ensName, disabled, onClick }: Wal
 
   return (
     <StyledButton onClick={() => onClick && onClick()} disabled={disabled}>
+      {address && <Davatar size={18} address={address} style={{ marginRight: 6 }} generatedAvatarType="jazzicon" />}
```
Comment (turtlemoji) [inline src/client/components/app/ConnectWalletButton/index.tsx:38]: Remove inline styles and use styled components

```suggestion
      {address && <StyledDavatar size={18} address={address} generatedAvatarType="jazzicon" />}
```

and then do 
```
const StyledDavatar = styled(Davatar)`
    margin-right: .6rem;
`;
```

**Comment by bigint** [comment]
@turtlemoji resolved all comments 🙇🏻 😄

**Comment by bigint** [comment]
@turtlemoji any updates on this?

**Review by FiboApe — APPROVED** [review]

**Comment by turtlemoji** [comment]
@bigint we didn't have the time to audit the dependencies yet

**Comment by FoxTheSin** [comment]
hey bigint, appreciate all the work both our side have done to get this in, but after discussion we've decided we want everything hard coded in our repositories (minimizing external dependencies for security reasons.)

If you'd like to start a new PR to hard code this in, the community would enjoy it. However we'll be closing this PR for external dependency security reasons.

**Comment by turtlemoji** [comment]
@bigint, tell me if you need any info with the new PR

**Pull request closed without merging**